src/tests/mod.rs

```diff
@@ -47,28 +53,47 @@ fn initialize() {
 /// * dest: Path to destination
 ///
 /// This method causes side effects - writes new files to a directory
-fn copy_folder(src: &Path, dst: &Path) {
-  for entry in fs::read_dir(src).unwrap() {
+fn copy_folder_to_temp_dir(src: &PathBuf) -> TempDir {
+  // Copy the test scenario to temporary directory
+  let temp_dir = TempDir::new_in(".", "tmp_test").unwrap();
+  let temp_dir_path = &temp_dir.path();
+  for entry in fs::read_dir(src.as_path()).unwrap() {
     let entry = entry.unwrap();
     let path = entry.path();
     if path.is_file() {
       _ = fs::copy(
         path.to_str().unwrap(),
-        dst.join(path.file_name().unwrap()).to_str().unwrap(),
+        temp_dir_path
+          .join(path.file_name().unwrap())
+          .to_str()
+          .unwrap(),
       )
       .unwrap();
     }
   }
+  temp_dir
 }
 
 /// Checks if the file updates returned by piranha are as expected.
-fn check_result(path_to_codebase: &Path, path_to_expected: &Path) {
+fn execute_piranha_and_check_result(
+  piranha_arguments: &PiranhaArguments, path_to_expected: &Path, files_changed: usize,
+  ignore_whitespace: bool,
+) {
+  let path_to_codebase = Path::new(piranha_arguments.path_to_codebase());
+  let output_summaries = execute_piranha(piranha_arguments);
+
+  assert_eq!(output_summaries.len(), files_changed);
+
   let mut all_files_match = true;
 
   let count_files = |path: &Path| {
     let mut count = 0;
-    for dir_entry in fs::read_dir(path).unwrap() {
-      if dir_entry.unwrap().path().is_file() {
+    for dir_entry in fs::read_dir(path).unwrap().flatten() {
+      if dir_entry.path().is_file() {
+        // If the directory contains a file named `.placeholder` we assume that the directory is empty
```

Comment:
Should we assume the directory is empty? Or just not count that particular file? Or check that the directory is otherwise empty and throw an exception if it isn't? The first of the three options seems like the most prone to silent errors...

Reply:
i think lets go with 2. ITs more intuitive.